crypto/http/http_lib.c

```diff
@@ -107,22 +126,24 @@ int OSSL_parse_url(const char *url, char **pscheme, char **puser, char **phost,
         p = ++host_end;
     } else {
         /* look for start of optional port, path, query, or fragment */
-        host_end = strchr(host, ':');
-        if (host_end == NULL)
-            host_end = strchr(host, '/');
-        if (host_end == NULL)
-            host_end = strchr(host, '?');
-        if (host_end == NULL)
-            host_end = strchr(host, '#');
+        host_end = ossl_strchr(host, ":/?#");
         if (host_end == NULL) /* the remaining string is just the hostname */
             host_end = host + strlen(host);
         p = host_end;
     }
 
     /* parse optional port specification starting with ':' */
     port = "0"; /* default */
-    if (*p == ':')
+    if (*p == ':') {
         port = ++p;
+    } else if (scheme) {
+        if (scheme_end - scheme == strlen("https")
+            && memcmp(scheme, "https", strlen("https")) == 0)
+            port = "443";
+        else if (scheme_end - scheme == strlen("http")
+                 && memcmp(scheme, "http", strlen("http")) == 0)
```

Comment:
Please use `sizeof("string") - 1` instead of strlen.

Reply:
I have now noticed that the handling of URLs without a specified port is already implemented in OSSL_HTTP_parse_url().
Therefore, I will keep the default port set to 0.